src/System.Configuration.ConfigurationManager/tests/System/Configuration/TempConfigurationHost.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Configuration;
+using System.Configuration.Internal;
+using System.Reflection;
+
+public class TempConfigurationHost : DelegatingConfigHost {
+    static string assemblyName = PlatformDetection.IsFullFramework ? "System.Configuration, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" : "System.Configuration.ConfigurationManager";
+    static IInternalConfigConfigurationFactory s_configurationFactory;
+    
+    private ConfigurationFileMap m_fileMap;
+
+    public TempConfigurationHost() {
+        Type type = Type.GetType(InternalHostTypeName, true);
+        Host = (IInternalConfigHost) Activator.CreateInstance(type, true);
+    }
+
+    public override void Init(IInternalConfigRoot configRoot, params object[] hostInitParams) {
+        Host.Init(configRoot, hostInitParams);
+    }
+
+    public override void InitForConfiguration(ref string locationSubPath, out string configPath, out string locationConfigPath,
+                        IInternalConfigRoot configRoot, params object[] hostInitConfigurationParams) {
+
+        Host.Init(configRoot, hostInitConfigurationParams);
+
+        m_fileMap = hostInitConfigurationParams[1] as ConfigurationFileMap;
+        
+        locationSubPath = ConfigurationFactory.NormalizeLocationSubPath(locationSubPath, null);
+        configPath = "MACHINE/EXE";
+        locationConfigPath = locationSubPath;
+
+        
+    }
+
+    public override bool IsTrustedConfigPath(string configPath) {
+        return true;
+    }
+
+    public override bool IsLocationApplicable(string configPath) {
+        return true;
+    }
+
+    public override bool IsFullTrustSectionWithoutAptcaAllowed(IInternalConfigRecord configRecord) {
+        return true;
+    }
+
+    public override bool PrefetchAll(string configPath, string streamName)
+    {
+        return false;
+    }
+
+    public override bool PrefetchSection(string sectionGroupName, string sectionName)
+    {
+        return false;
+    }
+
+    public override string GetStreamName(string configPath) {
+        return m_fileMap.MachineConfigFilename;
```

Comment:
If m_fileMap is being iniitalized with an `as`, that suggests it might be null, but then we dereference it here?

Reply:
In this case, IMCH does not use the filemap because it's given at a different position.  The test is set up to pass it at index 1 and that's what TempConfigurationHost uses.